internal/metrics/metrics.go

```diff
@@ -87,6 +88,8 @@ const (
 	statusUpdateConflict        = "contour_status_update_conflict_total"
 	statusUpdateNoop            = "contour_status_update_noop_total"
 	statusUpdateDurationSeconds = "contour_status_update_duration_seconds"
+
+	ContourCircuitBreakerSettings = "contour_dag_circuit_breaker_settings"
```

Comment:
Could you explain why you want to expose this as a metric? We don't really have any other metrics like this, so it's a little unusual.

Reply:
> Could you explain why you want to expose this as a metric? 

We want to alert and build graphs on this as a defence in depth. We have discovered during  a deployment that we accidentally deleted those annotations and that caused issues and that was really painful to debug.

Basically it helps with the discoverability of what people are setting across the cluster.

Reply:
Hm, I would expect something like this to be exposed via resource Status or something similar rather than a Contour xDS server metric

Also I believe we have an issue/PR to expand on cluster circuit breaker metrics: https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#circuit-breakers-statistics (here: https://github.com/projectcontour/contour/pull/5950)

I don't want to hold up the rest of the change, so maybe lets pull the metrics part out into another PR?

Reply:
> Hm, I would expect something like this to be exposed via resource Status or something similar rather than a Contour xDS server metric


Personally I take a more liberal approach to metrics in the sense that, if `contour` makes the decision to set a specific value it would be useful for operators to know the decision at runtime, especially if it affect things

Reply:
> I don't want to hold up the rest of the change, so maybe lets pull the metrics part out into another PR?

I won't die on this hill :D, So I  can separate the changes into two PRs. But my general take is that adding the metric is pretty "cheap" and overall useful. I.e. there have been cases where I wish I had this metric tracked.